src/composables/use-referral-info.ts

```diff
@@ -47,25 +60,70 @@ export const useReferralInfo = (poolId: number) => {
     }
   }
 
-  const getRefData = async (userAddress: string): Promise<ReferralData> => {
-    const [currentReward, multiplier, refData] = await Promise.all([
-      getCurrentReferrerReward(userAddress),
-      getReferrerMultiplier(userAddress),
-      getReferrersData(userAddress),
-    ])
-
-    const foundTier = tiers.value.find(
-      tier => tier.multiplier === multiplier.toString(),
+  const getDepositedAmountByUser = async (userAddress: string) => {
+    const depositedInfo = await fetchDepositedAmountUserReferrers(
+      poolId,
+      userAddress,
+      route.query.network as NETWORK_IDS,
+    )
+    refsCount.value = depositedInfo.length
+    return depositedInfo.reduce(
+      (acc: BigNumber, item: UserReferralDepositedAmount) => {
+        acc = acc.add(item.amount)
+        return acc
+      },
+      BigNumber.from(0),
     )
+  }
+
+  const findTierIndex = (amountStaked: ethers.BigNumber): number => {
+    let resultIndex = 0
+    tiers.value.forEach(tier => {
+      if (amountStaked.gte(BigNumber.from(tier.amount))) {
+        resultIndex = Number(tier.idx)
+      }
+    })
+    return resultIndex
+  }
+
+  const getRefData = async (userAddress: string): Promise<ReferralData> => {
+    const [currentReward, multiplier, refData, depositedAmount] =
+      await Promise.all([
+        getCurrentReferrerReward(userAddress),
+        getReferrerMultiplier(userAddress),
+        getReferrersData(userAddress),
+        getDepositedAmountByUser(userAddress),
+      ])
+
+    const foundTier = findTierIndex(refData.amountStaked)
 
     return {
+      depositedAmount,
       currentReward,
       multiplier,
-      tier: foundTier?.idx ?? '0',
+      tier: foundTier,
       ...refData,
     }
   }
 
+  const loadReferralDepositDataWithPagination = async (
+    referrer: string,
+    page: number,
+    limit: number,
+    order: SORTING_ORDER,
+    type: NETWORK_IDS,
+  ) => {
+    const skip = (page - 1) * limit
+    return fetchSpecificUserReferrers(
+      String(poolId),
+      referrer,
+      skip,
+      limit,
+      order,
+      type,
+    )
+  }
+
```

Comment:
Do you have `loadReferralDepositDataWithoutPagination` as well? 
If not just leave it as `loadReferralDepositData` or `getReferralDepositInfo`

Also divide arguments into
```ts
type PageParams = {
    number: number
    limit: number
    order: 'asc' | 'desc'
}

type ReferralDepositParams = {
   referrer: string
   type: NETWORK_IDS
}

const loadReferralDeposit = async (params: ReferralDepositParams, pageParams: PageParams)
```

Reply:
done